polygon/sync/store.go

```diff
@@ -179,6 +185,21 @@ func (s *ExecutionClientStore) insertBlocks(ctx context.Context, blocks []*types
 			"blks/sec", float64(len(blocks))/math.Max(time.Since(insertStartTime).Seconds(), 0.0001))
 	}
 
+	go func() {
```

Comment:
I don't recommend simply firing this goroutine without proper context handling, since this could lead to a goroutine leak if `insertBlocks` is cancelled for example. 

I would suggest something like this :

```
go func(ctx context.Context, blocks []*types.Block) {
		for i := range blocks {
			select {
			case <-ctx.Done():
				return
			default:
				wiggle, err := s.wiggleCalculator.CalculateWiggle(ctx, blocks[i].Header())
				if err != nil {
					s.logger.Error(
						syncLogPrefix("failed to update wiggle metrics"),
						"err", err,
					)
					continue
				}

				UpdateWiggleDuration(wiggle)
			}
		}
	}(ctx, blocks)
```

Also, think about if you want to parallelize the wiggle computation over the list of blocks (could be overkill, or could be useful depending on how heavy the wiggle computation is and how many blocks we are inserting at a time).